src/core/qgsvectorlayer.cpp

```diff
@@ -3029,6 +3070,16 @@ void QgsVectorLayer::updateFields()
     mFields[ index ].setConstraints( constraints );
   }
 
+  auto fieldWidgetIterator = mFieldWidgetSetups.constBegin();
```

Comment:
Why not in the first parameter of the for?
And last time I've used an auto, somebody told me we don't have C++11 yet on every platform. It's not the case anymore?